File: src/test/scala/net/igsoft/sdi/scala/testclasses/Stepper.scala
package net.igsoft.sdi.scala.testclasses

import scala.collection.mutable

class Stepper() {
  val steps : mutable.Buffer[String] = mutable.Buffer()

  def addStep(clazz: Class[_], step: String): Unit = {
    steps += clazz.getSimpleName + ":" + step
  }

  override def toString: String = steps.mkString(" ")
}

File: src/test/java/net/igsoft/sdi/testclasses/Stepper.java
package net.igsoft.sdi.testclasses;

import java.util.List;
import java.util.stream.Collectors;

import com.google.common.collect.Lists;

public class Stepper {
    private final List<String> steps;

    public Stepper() {
        this.steps = Lists.newArrayList();
    }

    public void addStep(Class<?> clazz, String step) {
        steps.add(clazz.getSimpleName() + ":" + step);
    }

    public List<String> getSteps() {
        return steps;
    }

    @Override
    public String toString() {
        return steps.stream().collect(Collectors.joining(" "));
    }
}
